test/OpenTelemetry.Exporter.Geneva.Tests/Testing.md

```diff
@@ -0,0 +1,188 @@
+user_events feature works.
+
+## Testing
+
+### Prerequisites
+
+* Perf Tool: Install the [perf](https://perf.wiki.kernel.org/index.php/Main_Page)
+  tool to capture the user_events.
+* Decode-Perf Tool: Install the [decode-perf](https://github.com/microsoft/LinuxTracepoints/tree/main/libeventheader-decode-cpp/tools)
+  tool to decode the user_events.
+
+### Steps for testing UnixUserEventsDataTransport
+
+To capture the user_events, the perf tool has to be run while `dotnet test` is
+running. The most simple way is to do in two terminals.
+
+#### Terminal 1
+
+First, remove `(Skip = "This would fail on Ubuntu. Skipping for now. See
+issue: #2326.")` for the `UserEvents_Logs_Success_Linux` test case to enable the
+tests.
+
+Go to the `test/OpenTelemetry.Exporter.Geneva.Tests/` folder and run the tests:
+
+```bash
+$ sudo dotnet test --configuration Debug --framework net8.0 --filter CategoryName=Geneva:user_events --no-build
+[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.8.2+699d445a1a (64-bit .NET 8.0.12)
+[xUnit.net 00:00:00.12]   Discovering: OpenTelemetry.Exporter.Geneva.Tests
+[xUnit.net 00:00:00.45]   Discovered:  OpenTelemetry.Exporter.Geneva.Tests
+[xUnit.net 00:00:00.46]   Starting:    OpenTelemetry.Exporter.Geneva.Tests
+[xUnit.net 00:00:00.51]     OpenTelemetry.Exporter.Geneva.Tests.UnixUserEventsDataTransportTests.UserEvents_Enabled_Success_Linux [SKIP]
+[xUnit.net 00:00:00.51]       This would fail on Ubuntu. Skipping for now.
+cat /sys/kernel/tracing/user_events_status
+cat /sys/kernel/tracing/user_events_status [OUT] Active: 0
+cat /sys/kernel/tracing/user_events_status [OUT] Busy: 0
+cat /sys/kernel/debug/tracing/trace_pipe
+cat /sys/kernel/tracing/events/user_events/MicrosoftOpenTelemetryLogs_L4K1/enable
+cat /sys/kernel/tracing/events/user_events/MicrosoftOpenTelemetryLogs_L4K1/enable [OUT] 0
+sh -c "echo '1' > /sys/kernel/tracing/events/user_events/MicrosoftOpenTelemetryLogs_L4K1/enable"
+------------- ready to write events -------------
+About to write tracepoint:
+Written tracepoint.
+cat /sys/kernel/debug/tracing/trace_pipe [OUT]   .NET TP Worker-19820   [004] ..... 620527.192577: MicrosoftOpenTelemetryLogs_L4K1: eventheader_flags=(7) version=(0) id=0x0 (0) tag=0x0 (0) opcode=(0) level=(4)
+Writing events from listener:
+eventheader_flags=(7), version=(0), id=0x0 (0), tag=0x0 (0), opcode=(0), level=(4)
+Total events: 1
+sh -c "echo '0' > /sys/kernel/tracing/events/user_events/MicrosoftOpenTelemetryLogs_L4K1/enable"
+[xUnit.net 00:00:10.63]     OpenTelemetry.Exporter.Geneva.Tests.UnixUserEventsDataTransportTests.UserEvents_Disabled_Success_Linux [SKIP]
+[xUnit.net 00:00:10.63]       This would fail on Ubuntu. Skipping for now.
+[xUnit.net 00:00:10.63]   Finished:    OpenTelemetry.Exporter.Geneva.Tests
+  OpenTelemetry.Exporter.Geneva.Tests test net8.0 succeeded (11.6s)
+
+Test summary: total: 3, failed: 0, succeeded: 1, skipped: 2, duration: 11.6s
+Build succeeded in 12.1s
+```
+
+#### Terminal 2
```

Comment:
The preview window isn't correctly parsing your numbered list here.

I think if you indent the code samples it might fix the auto-numbering.
Otherwise you would have to explicitly number these 1-5.

Reply:
I wish I could, but naming them 1-5 would result in error `MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 3; Style: 1/2/3] https://github.com/DavidAnson/markdownlint/blob/v0.37.4/doc/md029.md`.

Luckily indenting worked.